roles/iperf3-bench/tasks/main.yml

```diff
@@ -0,0 +1,54 @@
+---
+- block:
+
+  - include_role:
+      name: pre-result
+    when: store_results is defined and store_results
+
+  - name: Start Server(s)
+    k8s:
+      definition: "{{ lookup('template', 'server.yml.j2') | from_yaml }}"
+    with_sequence: start=1 count={{ iperf3.pairs }}
+
+  # TODO(aakarsh): check if all server pods are running
+  - name: Wait for pods to be running....
+    k8s_facts:
+      kind: Pod
+      api_version: v1
+      namespace: '{{ operator_namespace }}'
+      label_selectors:
+        - app = iperf3-bench-server
+    register: server_pods
+    until: "iperf3.pairs == (server_pods | json_query('resources[].status.podIP')|length)"
+    retries: 37
+    delay: 10
+
+  - name: Start Client(s)
+    k8s:
+      definition: "{{ lookup('template', 'client.yml.j2') | from_yaml }}"
+    with_items: "{{ server_pods.resources }}"
+    when: server_pods.resources|length > 0 and (store_results is not defined or (store_results is defined and not store_results))
+
+  - name: Start Client(s) - store results
+    k8s:
+      definition: "{{ lookup('template', 'client_store.yml.j2') | from_yaml }}"
+    with_items: "{{ server_pods.resources }}"
+    when: server_pods.resources|length > 0 and store_results is defined and store_results
+
+  - name: Waiting for Jobs to complete....
+    k8s_facts:
+      kind: Job
+      api_version: v1
+      namespace: '{{ operator_namespace }}'
+      label_selectors:
+        - app = iperf3-bench-client
+    register: client_jobs
+    until: "iperf3.pairs == (client_jobs | json_query('resources[].status.succeeded')|length)"
+    retries: 30
```

Comment:
If the user passes `transmit_type: time` and `transmit_value: 400` this will fail

Reply:
That was a smart catch, updated it to a max of 900 seconds.

Reply:
Well... I would suggest we check if transmit_type is time, and if it is time, it is ` {{ transmit_value + 300 }} ` for example... If transmit_type isn't time, the current logic is fine (not the 900 sec)